### ISSUE TITLE
Allow users to export databases

### ISSUE THREAD

**Issue by babyharpsea** [Jexactyl/Jexactyl#94]
### Is there an existing feature request for this?

- [X] I have searched the existing issues before opening this feature request.

### Describe the feature you would like to see.

A few users on my host suggested this feature, it would be great if you could import databases directly as it'd be way better for users moving within hosts. It'd also be great if the users could export the database into their server or download it directly, and if available make a cron job in order to do this automatically.

### Describe the solution you'd like.

In case of any security issues or such, a simple solution would be for the user to first input their account password, in case someone got in abusing a non-discovered security issue (cookies, such), and then asking for the database name, user, and password so it can be successfully exported without it exporting another database.

**Comment by camwhi-te** [comment]
Not sure how to approach this but a cool idea nonetheless.

**Comment by babyharpsea** [comment]
Is React/PHP able to send commands into the root server? If so, just export it via CLI and send the file to the user. If not, try to use MySQL I guess

**Comment by NotDemonix** [comment]
> Not sure how to approach this but a cool idea nonetheless.

Im thinking of stealing some code from file manager and somehow get to know where database file is located, MySQL store in .frm according to google so we can just allow users to upload .frm files and download them

**Comment by NotDemonix** [comment]
Tho im not good at PHP cuz im js dev, so i can only give ideas here, rest can be stolen from other places in code

**Comment by camwhi-te** [comment]
@babyharpsea That's probably how I'll end up doing things.

**Comment by camwhi-te** [comment]
Hmm, this only seems practical if the database is on the same system as Wings. I'll see what I can do in terms of this, but I can't make any promises.

**Comment by babyharpsea** [comment]
I've thought a bit more about this, and what if you do it so when a server gets transferred, it's DB's get transferred too? It's a pain manually moving each database

**Comment by babyharpsea** [comment]
What do you think @cmrxnn? Would this require any extra modification on the wings side?

**Comment by camwhi-te** [comment]
In general this seems like a relatively complicated procedure but there's nothing that can't be done. I'll do some testing today. I might not make a system to import, but exporting should be easy enough.

It could require a change on the Wings side, so that could mean i will have to maintain a fork of Wings also.

**Comment by babyharpsea** [comment]
Yeah, that's true. If you want I can maintain wings, currently learning Go

**Comment by 00kino547** [comment]
forgotten, but adding PMA, may work, has an import and expòrt button
